lib/req/request.ex

```diff
@@ -24,6 +24,9 @@ defmodule Req.Request do
 
     * `:error_steps` - the list of error steps
 
+    * `:location_trusted` - if `false`, authorization headers are not
+      sent to different hosts on redirects. Defaults to `false`.
```

Comment:
In the future we also want to control redirects with `:max_redirects` option (https://github.com/wojtekmach/req/issues/64) so instead of adding a new field to the struct, let's change `follow_redirects/1` to `follow_redirects/2` where the 2nd argument is options keyword list.

See how `retry` is configured in `put_default_steps` for a similar idea.

Reply:
so to be super clear, I think we'd have eg `Req.get(url, follow_redirects: [location_trusted: true])`. Now, I think what you had initially, `Req.get(url, location_trusted: true)` is actually quite nicer. But let's be consistent with the retry step in the meantime and if we want to change the approach to be more "flat", I'll do it in one go. (Sorry if this seems like a bit of an extra unnecessary work.)

Reply:
This was a little tricky since retry is optional and following redirects is not, but I used the same approach while not providing an option to not follow redirects (that could obviously be added later if needed).

Reply:
yeah, once we have max_redirects, one way to disable redirects is to set it to 0. `follow_redirects: false` would be another idea. Not sure which one is better. out of the scope of this PR anyway!